upload/views/legacy.py

```diff
@@ -259,25 +271,18 @@ def post(self, request, *args, **kwargs):
             )
 
             parse_headers(request.META, upload_params)
-            archive_service = ArchiveService(repository)
 
             # only Shelter requests are allowed to set their own `storage_path`
             path = upload_params.get("storage_path")
             if path is None or not self.is_shelter_request():
-                path = "/".join(
-                    (
-                        "v4/raw",
-                        timezone.now().strftime("%Y-%m-%d"),
-                        archive_service.get_archive_hash(repository),
-                        commitid,
-                        f"{reportid}.txt",
-                    )
-                )
+                path = default_path
 
             try:
-                upload_url = archive_service.create_raw_upload_presigned_put(
-                    commit_sha=commitid, filename="{}.txt".format(reportid)
-                )
+                # FIXME(swatinem): this is using the `default_path` even for the shelter path,
+                # how does that even work? shouldn't this be the same as the `path`?
+                # or is shelter rather making sure that the `upload_params`
+                # `storage_path` and `url` are set appropriately?
```

Comment:
do we need to answer this Q before merging?

Reply:
Working on getting an answer in https://sentry.slack.com/archives/C04NC9NNPEG/p1730968157331319?thread_ts=1730814495.218099&cid=C04NC9NNPEG
I believe when using shelter, it ignores the generated `upload_url` in favor of returning the `storage_path` back to the user which is then both user for the upload on the user end, as well as the read on the worker end.

Reply:
Yeah so to clarify cause this is confusing:

We have a client (cli, node (legacy) uploader, etc) talking to Codecov, and Codecov responds to them with a storage path and the client stores the report data directly in GCS. 

Before Shelter, the client would call API and API would create + respond that presigned URL back to the client, so client <-> API. After Shelter, the communication looks like client <-> shelter <-> api, so now shelter takes care of creating that storage_path and returning the presigned url back to the client, example [here](https://github.com/codecov/shelter/blob/a0291ac2d1ce562cc2f6f8571db5caa848191a27/upload-ingress/main.py#L230), so now the storage_path/url created in API is now returned to Shelter but Shelter does nothing with it. So every request that comes through Shelter takes care of this logic in Shelter side, so what you said about storage_path and url is correct.

Worth noting, Shelter does not take V2 uploads by choice, so any logic that is for V2 should remain here - i'm not as familiar w/ V2 logic so not sure how we deal with that. I see that we are now writing to V2 from API, have we always done that in worker in the first place?